File: processor/src/main/java/com/michaelrocks/lightsaber/processor/commons/JavaVersionChanger.java
/*
 * Copyright 2015 Michael Rozumyanskiy
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.michaelrocks.lightsaber.processor.commons;

import org.objectweb.asm.ClassVisitor;

import static org.objectweb.asm.Opcodes.ASM5;

public class JavaVersionChanger extends ClassVisitor {
    private final int newVersion;

    public JavaVersionChanger(final ClassVisitor classVisitor, final int newVersion) {
        super(ASM5, classVisitor);
        this.newVersion = newVersion;
    }

    @Override
    public void visit(final int version, final int access, final String name, final String signature,
            final String superName, final String[] interfaces) {
        super.visit(newVersion, access, name, signature, superName, interfaces);
    }
}

File: processor/src/main/java/io/michaelrocks/lightsaber/processor/commons/JavaVersionChanger.kt
/*
 * Copyright 2019 Michael Rozumyanskiy
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.michaelrocks.lightsaber.processor.commons

import org.objectweb.asm.ClassVisitor

import org.objectweb.asm.Opcodes.ASM5

class JavaVersionChanger(classVisitor: ClassVisitor, private val newVersion: Int) : ClassVisitor(ASM5, classVisitor) {
  override fun visit(
    version: Int,
    access: Int,
    name: String,
    signature: String?,
    superName: String?,
    interfaces: Array<String>?
  ) {
    super.visit(newVersion, access, name, signature, superName, interfaces)
  }
}
